js/post/update_post.js

```diff
@@ -1,50 +1,174 @@
+            setCookie(courseCode, 'inactive');
     }
-    */
+    updateAllCategories();
+});
+*/
 
-    var active = 0;
-    var inactive = 0;
-    
-    e.preventDefault();
-    var children = $(this).parent().children('.more-info').children();
-    for (i = 0; i < children.length; i++) {
-        
-        if (children[i].className.indexOf("CSC") != 0) {
-           var index = children[i].className.indexOf("CSC");
-        } else if (children[i].className.indexOf("Lin") != 0) {
-           var index = children[i].className.indexOf("Lin");
-        } else if (children[i].className.indexOf("Mat") != 0) {
-           var index = children[i].className.indexOf("Mat");
-        } 
+
+/**
+ * Updates all categories to see if they are fulfilled or not.
+**/
+// TODO: Add CSC240 and CSC265
+function updateAllCategories() {
+    'use-strict';
+
+    updateCompletedMinCourses();
+    updateCompletedMajCourses();
+    updateCompletedSpecCourses();
+    update300s();
+    update400s();
+    fill300s();
+    fill400s();
+    fillInquiries();
+    updateCreditCount();
+
+    // Update Specialist 
+    for (var property in completed_spec) {
+        if (completed_spec.hasOwnProperty(property)) {
+            var category = $('#spec_' + property.toLowerCase())[0].getElementsByClassName('code')[0];
+            if (completed_spec[property] === 1) { // if the category is completed
+                activateCourse(property);
+                updateCategory(category, 'fulfilled');
+            } else { // if the category is not completed
+                deactivateCourse(property);
+                updateCategory(category, 'not fulfilled');
+            }
+        }
+    }
+
+    // Update Major
+    for (var property in completed_maj) {
+        if (completed_maj.hasOwnProperty(property)) {
+            var category = $('#maj_' + property.toLowerCase())[0].getElementsByClassName('code')[0];
+            if (completed_maj[property] === 1) { // if the category is completed
+                activateCourse(property);
+                updateCategory(category, 'fulfilled');
+            } else { // if the category is not completed
+                deactivateCourse(property);
+                updateCategory(category, 'not fulfilled');
+            }
+        }
+    }
+
+    // Update Minor
+    for (var property in completed_min) {
+        if (completed_min.hasOwnProperty(property)) {
+            var category = $('#min_' + property.toLowerCase())[0].getElementsByClassName('code')[0];
+            if (completed_min[property] === 1) { // if the category is completed
+                activateCourse(property);
+                updateCategory(category, 'fulfilled');
+            } else { // if the category is not completed
+                deactivateCourse(property);
+                updateCategory(category, 'not fulfilled');
+            }
+        }
+    }
+
+    // Update 300s
+    var i = 0; 
+    var spec300s = $('.300lvlspec');
```

Comment:
In HTML5, I believe that classes and IDs need to start with an alphabetical character. Sometimes this will crash Firefox, too.